### PR TITLE
Umcp 123/change/payment method names

### PR DESCRIPTION
**Pull request by Ryouzanpaku**

### CODE REVIEW

File: view/frontend/web/css/unzer.css [hunk 1]
```diff
@@ -0,0 +1,5206 @@
+.unzerUI.divider {
```
Comment (sixer1182) [inline view/frontend/web/css/unzer.css:1]: Did you download the current one and adapted it?
Is there a way to include the one from static.unzer.com and just override the changes?

**Review by sixer1182 — CHANGES_REQUESTED** [review]
LGTM
Some findings need fixing.

File: README.md [hunk 2]
```diff
@@ -1,32 +1,34 @@
-[![Latest Version on Packagist](https://img.shields.io/packagist/v/heidelpay/magento2-merchant-gateway.svg?style=flat-square)](https://packagist.org/packages/heidelpay/magento2-merchant-gateway)
+[![Latest Version on Packagist](https://img.shields.io/packagist/v/unzerdev/magento2.svg?style=flat-square)](https://packagist.org/packages/unzerdev/magento2)
 [![PHP 7.1](https://img.shields.io/badge/php-7.1-blue.svg)](http://www.php.net)
 [![PHP 7.2](https://img.shields.io/badge/php-7.2-blue.svg)](http://www.php.net)
 [![PHP 7.3](https://img.shields.io/badge/php-7.3-blue.svg)](http://www.php.net)
```
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply README.md:4]: - I will add php 7.4.
- php 8 would actually need to be checked plugin wise.

File: view/frontend/web/css/unzer.css (hunk 1 above)
Comment (Ryouzanpaku) [reply view/frontend/web/css/unzer.css:1]: I am not completly sure how you mean this. I have overritten content and renamed it.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sixer1182) [reply README.md:4]: ok

File: Api/Checkout.php [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
+use Unzer\PAPI\Api\Data\Customer;
+use Unzer\PAPI\Helper\Order as OrderHelper;
 use Magento\Checkout\Model\Session;
 use Magento\Quote\Model\Quote;
 
 /**
  * Checkout API Interface Implementation.
  *
- * Copyright (C) 2019 heidelpay GmbH
+ * Copyright (C) 2021 Unzer GmbH
```
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply Api/Checkout.php]: fixed project wide.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Release Notes - Payment extension for Magento2 and heidelpay merchant gateway (MGW)
+# Release Notes - Payment extension for Magento2 and Unzer merchant gateway (PAPI)
```
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply CHANGELOG.md]: fixed project wide.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply README.md:4]: Added php 7.4 badge.

File: view/frontend/web/css/unzer.css (hunk 1 above)
Comment (sixer1182) [reply view/frontend/web/css/unzer.css:1]: ok fine

File: Model/Config.php [hunk 5]
```diff
@@ -26,31 +26,31 @@
...
-    const METHOD_INVOICE = 'hpmgw_invoice';
-    const METHOD_INVOICE_GUARANTEED_B2B = 'hpmgw_invoice_guaranteed_b2b';
-    const METHOD_INVOICE_GUARANTEED = 'hpmgw_invoice_guaranteed';
-    const METHOD_PAYPAL = 'hpmgw_paypal';
-    const METHOD_SOFORT = 'hpmgw_sofort';
+    const METHOD_BASE = 'unzer';
+    const METHOD_CARDS = 'unzer_cards';
+    const METHOD_DIRECT_DEBIT = 'unzer_direct_debit';
+    const METHOD_DIRECT_DEBIT_SECURED = 'unzer_direct_debit_secured';
+    const METHOD_FLEXIPAY_DIRECT = 'unzer_flexipay_direct';
```
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply Model/Config.php]: renamed it project wide.

File: etc/di.xml [hunk 6]
```diff
@@ -1,601 +1,601 @@
...
-            <argument name="validatorPool" xsi:type="object">HeidelpayMGWDirectDebitGuaranteedValidatorPool</argument>
-            <argument name="commandPool" xsi:type="object">HeidelpayMGWCaptureCommandPool</argument>
+            <argument name="valueHandlerPool" xsi:type="object">UnzerDirectDebitSecuredValueHandlerPool</argument>
+            <argument name="validatorPool" xsi:type="object">UnzerDirectDebitSecuredValidatorPool</argument>
+            <argument name="commandPool" xsi:type="object">UnzerCaptureCommandPool</argument>
         </arguments>
     </type>
 
-    <virtualType name="HeidelpayMGWFlexipayDirectConfig" type="Heidelpay\MGW\Model\Config">
+    <virtualType name="UnzerFlexipayDirectConfig" type="Unzer\PAPI\Model\Config">
```
[the remark this answers is not in the document]
Comment (sixer1182) [reply etc/di.xml:256]: this is still open

**Review by sixer1182 — COMMENTED** [review]
One change is missing

File: composer.json [hunk 7]
```diff
@@ -1,26 +1,26 @@
 {
     "name": "unzerdev/magento2",
-    "description": "This extension for Magento 2 provides a direct integration of the heidelpay payment types to your Magento 2 shop via the heidelpay merchant gateway (MGW).",
+    "description": "This extension for Magento 2 provides a direct integration of the Unzer payment types to your Magento 2 shop via the Unzer merchant gateway (PAPI).",
     "type": "magento2-module",
     "license": "Apache-2.0",
     "require": {
         "php": "~7.1.3|~7.2.0|~7.3.0|~7.4.0",
```
[the remark this answers is not in the document]
Comment (sixer1182) [reply composer.json:7]: ok, next time

File: i18n/en_US.csv [hunk 8]
```diff
@@ -27,21 +27,21 @@
...
 "Please transfer the amount of %1 to the following account after your order has arrived:<br /><br />Holder: %2<br/>IBAN: %3<br/>BIC: %4<br/><br/><i>Please use only this identification number as the descriptor: </i><br/>%5","Please transfer the amount of %1 to the following account after your order has arrived:<br /><br />Holder: %2<br/>IBAN: %3<br/>BIC: %4<br/><br/><i>Please use only this identification number as the descriptor: </i><br/>%5"
 
-"Cards","Cards"
-"Direct Debit","Direct Debit"
-"Direct Debit Guaranteed","Direct Debit (guaranteed)"
-"FlexiPay Direct","FlexiPay Direct"
+"Cards","Credit Card / Debit Card"
+"Direct Debit","SEPA Direct Debit"
+"Direct Debit Secured","SEPA Direct Debit secured"
+"FlexiPay Direct","Unzer Bank Transfer"
```
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply i18n/en_US.csv]: changed.

File: i18n/en_US.csv [hunk 9]
```diff
@@ -27,21 +27,21 @@
...
 "iDEAL","iDEAL"
 "Invoice","Invoice"
-"Invoice Guaranteed","Invoice Guaranteed"
-"Invoice Guaranteed B2B","Invoice Guaranteed B2B"
+"Invoice Secured","Invoice secured"
+"Invoice Secured B2B","IInvoice secured B2B"
```
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply i18n/en_US.csv]: changed.

File: Model/Config.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply Model/Config.php]: changed it, UI component though do only have the payment type "FlexiPayDirect".

File: i18n/en_US.csv [hunk 10]
```diff
@@ -1,16 +1,16 @@
...
-"HPMGW_ABOUT_US","heidelpay GmbH, shortly: heidelpay, is a leading payment institution for online payment methods authorized and regulated by BaFin, which offers the full range of services for an electronic payment processing from one single source: from processing of transactions and providing of acceptance contracts to monitoring and risk management.<br/><br/>For more information please visit <a href='https://www.heidelpay.com/en/integration/shopmodules/' target='_blank'>www.heidelpay.com/en/</a>"
+"UNZER_PAYMENT","<img src='data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAMxJREFUeNpiZBXzZSAFMDGQCFhwSXBysAX6WPl7WhgbqMhKi3789E1MLRK7BiYmxvQEr5qSCFFh/rMX7qzbcuzOveePnrzCboOgAM+KuRWONnor1h1s6V1x685TfE4Cqt63sR3oAP/opu17zhDwA9AlQLOBqh18y69cf0jY00B3A10CNBuPakSwAsME6MuV6w/hcgm6BmAIAsOktXclsREHDO/zl+7euP2YWA3A2Dl0/AoJSQMYOPcfviBBAzDm7z98SYwGRpqnVoAAAwCge0cwGsIpTQAAAABJRU5ErkJggg==' style='margin-right: 5px'/>Unzer Payment (PAPI)"
```
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply i18n/en_US.csv]: changed.

File: composer.json [hunk 11]
```diff
@@ -1,26 +1,26 @@
...
-        "heidelpay/heidelpay-php": ">=1.2.7.0 <1.3.0.0",
+        "unzerdev/php-sdk": "~1.0",
```
[the remark this answers is not in the document]
Comment (sixer1182) [reply composer.json:8]: what about this?

File: composer.json [hunk 12]
```diff
@@ -1,26 +1,26 @@
...
     "autoload": {
         "files": [
             "registration.php"
         ],
         "psr-4": {
-            "Heidelpay\\MGW\\": ""
+            "Unzer\\PAPI\\": ""
         }
     },
     "homepage": "https://dev.unzer.com/user-guide-magento-2-mgw-plugin",
```
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply composer.json:26]: yes, I will update it later, when I work on ducomentation

File: README.md [hunk 13]
```diff
@@ -41,17 +43,17 @@ See the License for the specific language governing permissions and
 limitations under the License.
 
 ## User Guide
-Please find information on installation, configuration and usage here https://dev.heidelpay.com/user-guide-magento-2-mgw-plugin
+Please find information on installation, configuration and usage here https://dev.unzer.com/user-guide-magento-2-mgw-plugin
```
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply README.md:47]: will update, when I work on documentation.

File: composer.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (Ryouzanpaku) [reply composer.json]: done now.

File: README.md [hunk 14]
```diff
@@ -28,7 +29,7 @@ It also depends on the unzerdev/php-sdk, which will be installed along with the
 Please refer to the additional requirements listed under https://docs.unzer.com/docs/system-requirements (et seq.).
 
 ## LICENSE
-Copyright (C) 2019 Unzer GmbH
+Copyright (C) 2021 Unzer GmbH
```
Comment (sixer1182) [inline README.md:32]:
```suggestion
Copyright (C) 2021 - today Unzer GmbH
```

**Review by sixer1182 — APPROVED** [review]
LVGTM

**Pull request merged**